src/routers/index.ts

```diff
@@ -11,11 +11,13 @@ import { proposalsRouter } from './proposalsRouter';
 import { proposalVersionsRouter } from './proposalVersionsRouter';
 import { usersRouter } from './usersRouter';
 import { documentationRouter } from './documentationRouter';
+import { baseFieldLocalizationsRouter } from './baseFieldLocalizationsRouter';
 
 const rootRouter = express.Router();
 
 rootRouter.use('/applicationForms', applicationFormsRouter);
 rootRouter.use('/baseFields', baseFieldsRouter);
+rootRouter.use('/baseFieldLocalizations', baseFieldLocalizationsRouter);
```

Comment:
Rather than routes like `PUT /baseFieldLocalizations` and `POST /baseFieldLocalizations` with the base field ID in the body of the request, I think we want routes like `PUT /baseFields/:id/localizations/:lang` and `POST /baseFields/:id/localizations/`.

CC @slifty on this API design question.

Reply:
I do like the concept (in my head I said "oooooo nice!")

*but* we already have an API design that treats nested entities as root entities (e.g. it isn't `POST /proposals/:id/version/:number` but rather is `POST /proposalVersions` even though a version requires a proposal)

For consistency I think we may want to keep the simpler / less semantically meaningful URL patterns for now.

Reply:
I agree that the structure @jasonaowen laid out makes more sense semantically, mostly because updating a base field localization should be done in the context of the base field it relates to. However, since the pattern already exists, maybe we refactor them both in a later commit?

Reply:
Or we could start a new pattern, and fix the existing violations later? "We're doing it wrong elsewhere" is a depressing argument against doing it right here.

Reply:
@jasonaowen true. If we are deciding to nest the router and adopt this pattern, I'd be happy to create an issue for and tackle a refactor commit for the proposalVersions routes

Reply:
I do think it opens a can of worms -- it's less "we do this elsewhere" and more "our entire API design is flat right now and this would reflect a change that warrants a consideration of a larger refactors overall"

It isn't just the example I gave, it is almost everything.

Opportunities -> Application forms
Application Forms -> Application form fields 
Proposals -> Proposal Versions
Proposal Versions -> Proposal Field Values

The first question I'd ask is: what is the heuristic we use to decide that something should be a nested relationship vs a primary entity with relationship conditions?

The second is: is this PR the place we should create that heuristic, or should we merge this feature with the existing project pattern and create an issue to refactor more comprehensively?

I'm not against the refactor as a concept, but I think that implementing something differently because it feels right here without taking the time to think more comprehensively is a recipe for potential thrash.

Reply:
(Whoops, seeing below the "let's not burden harry's PR with this discussion" sorry!  +1 to chatting through it all by voice next week!)

Reply:
@slifty, @bickelj, @hminsky2002, and I discussed this further in a call today. Here's my understanding of where we landed; please either give a :+1: or reply with corrections!

- the endpoints will be like `/baseFields/:baseFieldId/:langTag`
- the table will still have a natural key on `(base_field_id, lang_tag)` with a foreign key on `base_field_id`
- I'll open up an issue for a larger discussion of dropping the base field id surrogate key in favor of the short code natural key
- we'll all reflect on nested routes and maybe talk about it more someday

Thanks for the engaging conversation! Does this sound right to everyone? Did I miss anything?

Reply:
Yeah that's a solid summary!  Thanks for opening the issue for moving to natural keys for base fields more generally.